courses/models.py

```diff
@@ -200,7 +200,7 @@ def next_run_date(self):
             (
                 course_run.start_date
                 for course_run in self.courseruns.all()
-                if course_run.start_date > now
+                if course_run.live and course_run.start_date > now
```

Comment:
can we add these condition in queryset filter?
e.g (untested though)

`self.courseruns.filter(live=True, start_date__gt=now)`

Reply:
There's a note on top of this block that deals specifically with this: it is done this way in order to leverage the caching of `prefetch_related`. Adding `filter` here would break that caching capability so we're doing the filtering in code here.